pykwasm/src/pykwasm/kdist/wasm-semantics/numeric.md

```diff
@@ -115,9 +115,9 @@ module WASM-NUMERIC
 `*UnOp` takes one oprand and returns a `Val`.
 
 ```k
-    syntax Val ::= IValType "." IUnOp   Int   [klabel(intUnOp)    , function]
+    syntax Val ::= IValType "." IUnOp   Int   [klabel(intUnOp)    , function, total]
```

Comment:
Hmmm... Is this really a total function? Let's take the following rule:
```
rule ITYPE . clz    I1 => < ITYPE > #width(ITYPE) -Int #minWidth(I1)
```
which calls `#minWidth(I1)`, which is _NOT_ marked as total and which is defined by the following two rules:
```
    rule #minWidth(0) => 0
    rule #minWidth(N) => 1 +Int #minWidth(N >>Int 1)                                 requires N =/=Int 0
```
which, I'm guessing, will create an infinite loop when `N <Int 0`. If this is correct, then `IValType "." IUnOp Int` is not defined when applied to a type, `clz` and a negative number. Even more, it can't even be a total function for which we didn't write all rules.

So, could you check to see what happens when applied to `clz` and a negative number?

Reply:
Hmm, that seems more like a problem with `#minWidth` than with the `clz` instruction (which I would think should be total in one way or the other, as it needs to have a result in WASM anyway, right?).
But `N >>Int 1` is still going to reduce `N` to zero eventually, even if `N` is negative, so it might still be OK?

Would it be safe to add a rule
```
   rule ITYPE . clz I1 => 0 requires I1 <Int 0
```
(with the rationale that `I1` in 2-s complement must start with a `1` if it is negative)